### PR TITLE
lzip: 1.23 -> 1.24

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

Changes in version 1.24:

   * The option '--empty-error', which forces exit status 2 if any empty 
member is found, has been added.

   * The option '--marking-error', which forces exit status 2 if the first 
LZMA byte is non-zero in any member, has been added.

   * File diagnostics have been reformatted as 'PROGRAM: FILE: MESSAGE'.

   * Diagnostics caused by invalid arguments to command-line options now 
show the argument and the name of the option.

   * The option '-o, --output' now preserves dates, permissions, and 
ownership of the file when (de)compressing exactly one file.

   * The option '-o, --output' now creates missing intermediate directories 
when writing to a file.

   * The variable MAKEINFO has been added to configure and Makefile.in.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/lz/lzip/package.nix [hunk 1]
```diff
@@ -7,12 +7,13 @@
...
   pname = "lzip";
-  version = "1.23";
+  version = "1.24";
   outputs = [ "out" "man" "info" ];
 
   src = fetchurl {
     url = "mirror://savannah/lzip/${pname}-${version}.tar.gz";
-    sha256 = "sha256-R5LAR93xXvKdVbqOaKGiHgy3aS2H7N9yBEGYZFgvKA0=";
+    sha256 = "d42659229b10e066eeb6e81eb673cdd893b672e512d26719c2d95975556ca56c";
+    # hash from release announcement
```
Comment (drupol) [inline pkgs/by-name/lz/lzip/package.nix:16]: IMHO, I don't think this comment is necessary here.

Comment (ehmry) [reply pkgs/by-name/lz/lzip/package.nix:16]: > IMHO, I don't think this comment is necessary here.

I don't think this comment is necessary here.

Comment (drupol) [reply pkgs/by-name/lz/lzip/package.nix:16]: Hey hey nice one !

Comment (ehmry) [reply pkgs/by-name/lz/lzip/package.nix:16]: Seriously though, having providence information is better than "nix build gave me this hash".

Comment (kirillrdy) [reply pkgs/by-name/lz/lzip/package.nix:16]: both arguments are valid, let's reach a conclusion while being respectful and polite

File: pkgs/by-name/lz/lzip/package.nix [hunk 2]
```diff
@@ -7,12 +7,13 @@
 
 stdenv.mkDerivation rec {
```
Comment (teto) [inline pkgs/by-name/lz/lzip/package.nix:8]: maybe use finalAttrs scheme instead of rec

**Comment by drupol** [comment]
Thanks for your hard work on this, closing the issue, [r-ryantm](https://github.com/r-ryantm) fixed it!

https://github.com/NixOS/nixpkgs/pull/284905

**Pull request closed without merging**